build.sh

```diff
@@ -25,226 +26,178 @@ fi
-           ;;
-    esac
+	case "$flag" in
+	h)
+		usage
+		exit 0
+		;;
+	u)
+		UPLOAD="YES"
+		;;
+	k)
+		AUTHKEYFILE="$(realpath "$OPTARG")"
+		;;
+	*)
+		exit_error "$(usage)"
+		;;
+	esac
 done
 shift "$((OPTIND-1))"
 
-# Set default values if not set
-# XXX: this might be unnecessary
-if [ -z "${AUTHKEYFILE}" ]; then
-    touch authorized_keys_in
-    AUTHKEYFILE="authorized_keys_in"
-fi
-
 # General Variables
 BASEDIR="$PWD"
 CDMOUNT="cd-rom"
-CHECKMOUNTCD1="$(mount | { grep "${CDMOUNT}" || :; } | awk '{print $1}')"
+CHECKMOUNTCD1="$(mount | { grep "$CDMOUNT" || :; } | awk '{print $1}')"
 FREEBSDISOSRC="https://download.freebsd.org/releases/amd64/amd64/ISO-IMAGES/13.1/FreeBSD-13.1-RELEASE-amd64-dvd1.iso"
 FREEBSDISOFILE="FreeBSD-13.1-RELEASE-amd64-dvd1.iso"
 MFSBSDDIR="mfsbsd"
 MYRELEASE="13.1-RELEASE"
 MYARCH="amd64"
-OUTIMG="mfsbsd-${MYRELEASE}-${MYARCH}.img"    # not in use
-OUTISO="mfsbsd-${MYRELEASE}-${MYARCH}.iso"    # in use
-MYBASE="${BASEDIR}/${CDMOUNT}/usr/freebsd-dist"
-MYCUSTOMDIR="${BASEDIR}/customfiles"
+OUTIMG="mfsbsd-$MYRELEASE-$MYARCH.img"    # not in use
+OUTISO="mfsbsd-$MYRELEASE-$MYARCH.iso"    # in use
+MYBASE="$BASEDIR/$CDMOUNT/usr/freebsd-dist"
+MYCUSTOMDIR="$BASEDIR/customfiles"
 
 # make sure we're in base directory
-cd "${BASEDIR}" || exit
+cd "$BASEDIR" || exit
 
 # check remote settings
-if [ -f "${BASEDIR}/settings.sh" ]; then
-    # shellcheck source=/dev/null
-    source "${BASEDIR}/settings.sh"
+if [ -f "$BASEDIR/settings.sh" ]; then
+	# shellcheck source=customfiles/depenguin_settings.sh.sample
+	source "$BASEDIR/settings.sh"
 fi
 
-if [ -z "${CFG_SSH_REMOTEHOST}" ]; then
-    CFG_SSH_REMOTEHOST=depenguin-me
+if [ -z "$CFG_SSH_REMOTEHOST" ]; then
+	CFG_SSH_REMOTEHOST=depenguin-me
 fi
 
 # create directory if not existing
-if [ ! -d "${BASEDIR}/${CDMOUNT}" ]; then
-    echo "creating ${CDMOUNT} directory"
-    mkdir -p "${BASEDIR}/${CDMOUNT}"
+if [ ! -d "$BASEDIR/$CDMOUNT" ]; then
+	echo "creating $CDMOUNT directory"
+	mkdir -p "$BASEDIR/$CDMOUNT"
 fi
 
 # unmount any existing loopback mount
-if [ -n "${CHECKMOUNTCD1}" ]; then
-    umount "${CHECKMOUNTCD1}"
+if [ -n "$CHECKMOUNTCD1" ]; then
+	umount "$CHECKMOUNTCD1"
 fi
 
 # fetch the iso
-if [ ! -f "${BASEDIR}/${FREEBSDISOFILE}" ]; then
-    fetch "${FREEBSDISOSRC}" -o "${BASEDIR}/${FREEBSDISOFILE}"
+if [ ! -f "$BASEDIR/$FREEBSDISOFILE" ]; then
+	fetch "$FREEBSDISOSRC" -o "$BASEDIR/$FREEBSDISOFILE"
 fi
 
 # mount the iso file
-if [ -f "${BASEDIR}/${FREEBSDISOFILE}" ]; then
-    mount -t cd9660 /dev/"$(/sbin/mdconfig -f "${FREEBSDISOFILE}")" "${BASEDIR}/${CDMOUNT}"
+if [ -f "$BASEDIR/$FREEBSDISOFILE" ]; then
+	mount -t cd9660 /dev/"$(
+	  /sbin/mdconfig -f "$FREEBSDISOFILE")" "$BASEDIR/$CDMOUNT"
 fi
 
 # check for git submodule dir mfsbsd, change directory into it
-if [ -d "${MFSBSDDIR}" ]; then
-    cd "${MFSBSDDIR}" || exit
+if [ -d "$MFSBSDDIR" ]; then
+	cd "$MFSBSDDIR" || exit
 fi
 
 # clean any prior builds
 make clean
 
-# copy in our custom configs
```

Comment:
Below I replaced repetitive code with arrays and loops.

I also added removing files in case they are not in custom (this way removing a custom file should have an effect in the build as well).